flax/optim/base.py

```diff
@@ -479,18 +493,18 @@ def __init__(
   def init_state(self, params):
     sub_states = []
     seen = {}
+    dummies = traverse_util.unflatten_dict({
```

Comment:
Is it intentional to still use flatten_dict here? It's nice for the error message but it still breaks non-dict optimizer targets this way.

Reply:
Oh - I was only thinking of `ModelParamTraversal`...

PTAL, should work with arbitrary traversals now (incl. useful error message - assuming containers know how to `__str__`).